libraries/botbuilder/src/botFrameworkAdapter.ts

```diff
@@ -1357,6 +1357,19 @@ export class BotFrameworkAdapter extends BotAdapter implements ExtendedUserToken
         return response;
     }
 
+    public async healthCheck(context: TurnContext) : Promise<HealthCheckResponse> {
+        const healthResults = <HealthResults>{
+            success: true,
+            "user-agent": USER_AGENT,
+            messages: [ 'Health check succeeded.' ]
+        };
+        if (this.credentials.appId) {
+            const token = await this.credentials.getToken();
+            healthResults.authorization = `Bearer ${ token }`;
+        }
+        return { healthResults: healthResults };
```

Comment:
```suggestion
if (!(await this.credentialProvider.isAuthenticationDisabled())) {
            const credentials = context.turnState.get(ConnectorClientCacheKey).credentials ||  this.credentials;
            const token = await this.credentials.getToken();
            healthResults.authorization = `Bearer ${ token }`;
        }
        return { healthResults: healthResults };
```

Reply:
OK updated now. (And you almost got it right - I assume you intended to use the symbol for the cache key and the local credentials on line 1373 😃)

Thanks for the review - this is better - and actually slightly closer to the C#.